src/main/java/uk/gov/ons/ctp/response/collection/exercise/service/impl/EventValidator.java

```diff
@@ -50,6 +53,74 @@ public boolean validate(
     return validateMandatoryEvents(eventMap) && validateNonMandatoryEvents(eventMap);
   }
 
+  /** Validates the dates on event creation */
+  public boolean validateOnCreate(
+      final List<Event> existingEvents,
+      final Event newEvent,
+      final CollectionExerciseDTO.CollectionExerciseState collectionExerciseState) {
+    Map<String, Event> events =
+        existingEvents.stream().collect(Collectors.toMap(Event::getTag, Function.identity()));
+    if (collectionExerciseState.equals(CollectionExerciseDTO.CollectionExerciseState.CREATED)) {
+      return validateMandatoryEventsOnCreate(events, newEvent);
+    }
+    return false;
+  }
+
+  private boolean validateMandatoryEventsOnCreate(
```

Comment:
The different test combinations I see for a BusinessEventValidator are any combination of the 4 mandatory events, these are the events already set for a CollectionExercise and are passed in the eventMap. So that is a total of 16 valid combinations of existing event. Then the new Event which can be one of the 4 mandatory events and a null. So I can only identify 65 tests. I will change the logic to prioritise the new Event on selection so this will work for update events also.